src/pds/web_analytics/s3_sync.py

```diff
@@ -0,0 +1,427 @@
+
+        # Upload files
+        uploaded_count = 0
+        total_files = 0
+
+        for root, _dirs, files in os.walk(src_path):
+            for file in files:
+                file_path = os.path.join(root, file)
+
+                if self.should_upload_file(file_path, all_patterns):
+                    total_files += 1
+
+                    # Calculate S3 key
+                    rel_path = os.path.relpath(file_path, src_path)
+                    s3_key = os.path.join(s3_base_path, rel_path).replace("\\", "/")
+
+                    print(f"Uploading: {file_path} -> s3://{self.bucket_name}/{s3_key}")
+
+                    if self.upload_file(file_path, s3_key):
+                        uploaded_count += 1
+
+                        # Delete source file if requested
+                        if self.delete:
+                            try:
+                                os.remove(file_path)
+                                print(f"Deleted source file: {file_path}")
+                            except Exception as e:
+                                print(f"Error deleting source file {file_path}: {str(e)}")
+
+        if uploaded_count > 0:
+            print(f"{src_path} sync to {s3_base_path}: {uploaded_count}/{total_files} files uploaded successfully.")
+        else:
+            print(f"{src_path} sync to {s3_base_path}: no files to upload.")
+
+    @staticmethod
+    def convert_size(size: int) -> str:
+        """Convert a size in bytes to a human-readable string.
+
+        Args:
+            size (int): The size in bytes.
+
+        Returns:
+            str: The human-readable size.
+        """
+        if size == 0:
+            return "0B"
+        size_name = ("B", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB")
+        i = int(math.floor(math.log(size, 1024)))
+        p = math.pow(1024, i)
+        s = round(size / p, 2)
+        return f"{s}{size_name[i]}"
+
+    @staticmethod
+    def get_bytes(size: int, size_incre: str) -> int:
+        """Convert a size with unit to bytes.
+
+        Args:
+            size (int): The numerical part of the size.
+            size_incre (str): The unit of the size (e.g., "KiB", "MiB").
+
+        Returns:
+            int: The size in bytes.
+        """
+        # Convert size to bytes based on unit
+        if size_incre == "KiB":
+            size *= 2**10
+        elif size_incre == "MiB":
+            size *= 2**20
+        elif size_incre == "GiB":
+            size *= 2**30
+        elif size_incre == "TiB":
+            size *= 2**40
+        return size
+
+    @staticmethod
+    def get_throughput(sent: int, start_time: float) -> str:
+        """Calculate and return the data transfer throughput.
+
+        Args:
+            sent (int): The number of bytes sent.
+            start_time (float): The start time of the transfer.
+
+        Returns:
+            str: The throughput in MB/s.
+        """
+        elapsed_time = time.monotonic() - start_time
```

Comment:
This only works if `start_time` itself is set with a call to `time.monotonic()`; maybe a warning in the docstring about this?

Reply:
added a comment.